### PR TITLE
Skip test_vault_reload with enabling/disabling mlock in LXD

### PR DESCRIPTION
**Pull request by nobuto-m**
When Vault is running in a LXD container, test_vault_reload is expected
to fail because mlock cannot be enabled in LXD containers. Skip it when
running in containers is detected. It's useful for local development
environments with Juju's localhost/LXD provider. The test won't be
skipped in the OpenStack Charm CI since the Vault units will be run in
KVM, not in LXD containers.

### CODE REVIEW

File: zaza/openstack/charm_tests/vault/tests.py [hunk 1]
```diff
@@ -277,6 +278,15 @@ def test_vault_reload(self):
             raise unittest.SkipTest("The version of charm-vault tested does "
                                     "not have reload action")
 
+        container_results = zaza.model.run_on_leader(
+            "vault", "systemd-detect-virt --container"
+        )
+        container_rc = json.loads(container_results["Code"])
```
Comment (gnuoy) [inline zaza/openstack/charm_tests/vault/tests.py:284]: Could you use `zaza.utilities._local_utils.get_provider_type` here instead ?

Comment (nobuto-m) [reply zaza/openstack/charm_tests/vault/tests.py:284]: Thanks, I didn't know about `get_provider_type`.

It returns `None` with my environment, and there is no clouds.yaml neither on the machine using for localhost/LXD provider or another machine with AWS provider. I'm not sure we can rely on get_provider_type without having clouds.yaml here.

```
2021-06-13 09:40:58 [INFO] Using config /home/nobuto/.local/share/juju/clouds.yaml
None
```

```
$ ls -1F ~/.local/share/juju/
accounts.yaml
bootstrap-config.yaml
controllers.yaml
cookies/
credentials.yaml
lxd/
models.yaml
public-clouds.yaml
ssh/
```

Comment (gnuoy) [reply zaza/openstack/charm_tests/vault/tests.py:284]: ok, lets go ahead with the patch as it is then

**Review by gnuoy — APPROVED** [review]
LGTM, thanks

**Pull request merged**